File: 2486-append-characters-to-string-to-make-subsequence/2486-append-characters-to-string-to-make-subsequence.java
class Solution {
    public int appendCharacters(String s, String t) {
        int i = 0, j = 0;
        int sn = s.length(), tn = t.length();
        while (i < sn && j < tn) {
            if (s.charAt(i) == t.charAt(j)) {
                i++;
                j++;
            } else {
                i++;
            }
        }
        return tn - j;
    }
}

File: LeetCode/2486-append-characters-to-string-to-make-subsequence/2486-append-characters-to-string-to-make-subsequence.py
class Solution:
    def appendCharacters(self, s: str, t: str) -> int:
        i, j = 0, 0
        sn, tn = len(s), len(t)
        while i < sn and j < tn:
            if s[i] == t[j]:
                j += 1
            i += 1
        return tn - j